### ISSUE TITLE
Improper Memory Management in RCComponent

### ISSUE THREAD

**Issue by abhinavkrin** [RocketChat/EmbeddedChat#157]
### Description:
In `src/index.js` , line *43*
```
const RCInstance = new RocketChatInstance(host, roomId);
```
RocketChatInstance should not be instantiated in this way, this will lead to the creation of an instance each time the component is rendered. 
1. This could lead to multiple connections or connecting rc server and fetching messages on each render. This should not happen when props like `roomId` and `host` are not changing. 
2. Unnecessary renders sinch RCInstance is passed through a Context.Provider.

### Steps to reproduce:
Self reproducible

### Expected behavior:
RCInstance should be instantiated using useMemo and should also be added to dependencies of useEffects wherever required
```
const RCInstance = useMemo(() => {
    new RocketChatInstance(host, roomId);
}, [host, roomId]);
```

### Actual behavior:

I stored each instance in a global array.
![ec_use_memoization](https://user-images.githubusercontent.com/15830206/221730856-04dfe2d8-b164-481c-98d6-eabd1ad098cd.png)

Then I tried sending messages with each.  I was able to send messages from all of them. 
![ec_memory_leak](https://user-images.githubusercontent.com/15830206/221730477-541e8afe-7060-4d39-9ff8-41b8a03c2e8f.png)

**Comment by MahirPatel3130** [comment]
is anyone working on this bug ?

**Comment by abhinavkrin** [comment]
This issue was closed by #161
